From b08a698aefc99b3de47601e39c2d654492e3611d Mon Sep 17 00:00:00 2001
From: nsamsono <nsamsono@marvell.com>
Date: Tue, 24 Oct 2017 21:25:25 +0300
Subject: [PATCH 5/5] ethdev: fix port detach function

Workaround for detach vdev according driver name as it
was done in attach vdev.

In DPDK 17.11 version (v17.11-rc1 tag) this issue is already fixed.

Change-Id: Ica51d4010ddbb8259a6d17fb3ff47d8e7be7efb9
Signed-off-by: nsamsono <nsamsono@marvell.com>
---
 lib/librte_ether/rte_ethdev.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/librte_ether/rte_ethdev.c b/lib/librte_ether/rte_ethdev.c
index 83898a8..fe2f5ca 100644
--- a/lib/librte_ether/rte_ethdev.c
+++ b/lib/librte_ether/rte_ethdev.c
@@ -438,8 +438,8 @@ struct rte_eth_dev *
 	if (rte_eth_dev_is_detachable(port_id))
 		goto err;
 
-	snprintf(name, sizeof(rte_eth_devices[port_id].data->name),
-		 "%s", rte_eth_devices[port_id].data->name);
+	snprintf(name, sizeof(rte_eth_devices[port_id].data->drv_name),
+		 "%s", rte_eth_devices[port_id].data->drv_name);
 	ret = rte_eal_dev_detach(name);
 	if (ret < 0)
 		goto err;
-- 
1.9.1

